app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java

```diff
@@ -365,26 +377,40 @@ private void setUsername(String name) {
             username.setText(usernameText);
         }
 
-        private void setCreatedAt(@Nullable Date createdAt) {
-            // This is the visible timestampInfo.
-            String readout;
-        /* This one is for screen-readers. Frequently, they would mispronounce timestamps like "17m"
-         * as 17 meters instead of minutes. */
-            CharSequence readoutAloud;
-            if (createdAt != null) {
-                long then = createdAt.getTime();
-                long now = new Date().getTime();
-                readout = DateUtils.getRelativeTimeSpanString(timestampInfo.getContext(), then, now);
-                readoutAloud = android.text.format.DateUtils.getRelativeTimeSpanString(then, now,
-                        android.text.format.DateUtils.SECOND_IN_MILLIS,
-                        android.text.format.DateUtils.FORMAT_ABBREV_RELATIVE);
+        protected void setCreatedAt(@Nullable Date createdAt) {
+            if (useAbsoluteTime) {
+                String time = "ERROR!";
+                if (createdAt != null) {
+                    SimpleDateFormat sdf;
+                    if (new Date().getTime() - createdAt.getTime() > 86400000L) {
+                        sdf = new SimpleDateFormat("MM/dd HH:mm:ss", Locale.getDefault());
+                    } else {
+                        sdf = new SimpleDateFormat("HH:mm:ss", Locale.getDefault());
+                    }
+                    time = sdf.format(createdAt);
+                    timestampInfo.setText(time);
+                }
```

Comment:
there is no `else` branch for when the `createdAt` is null (just set the timestamp to ? or something)